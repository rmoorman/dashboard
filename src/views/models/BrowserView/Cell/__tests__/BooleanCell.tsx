import * as React from 'react' // tslint:disable-line
import BooleanCell from '../BooleanCell'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

test('BooleanCell renders', () => {

  const save = jest.fn((value: TypedValue) => { /* */ })
  const cancel = jest.fn()
  const onKeyDown = jest.fn()
  const field = {
    id: 'cip3p48sj001e1jsmghwkdt2k',
    name: 'description',
    description: '',
    isReadonly: true,
    isList: false,
    isSystem: true,
    typeIdentifier: 'Boolean',
    relatedModel: null,
  }

  const component = shallow(
    <BooleanCell
      value={false}
      save={save}
      cancel={cancel}
      onKeyDown={onKeyDown}
      field={field}
    />
  )

  expect(shallowToJson(component)).toMatchSnapshot()

})
