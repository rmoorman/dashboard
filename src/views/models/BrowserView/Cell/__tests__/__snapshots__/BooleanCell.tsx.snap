exports[`test BooleanCell renders 1`] = `
<ToggleButton
  leftText="false"
  onClickOutside={[Function]}
  onKeyDown={[Function]}
  rightText="true"
  side={0} />
`;
